custom_components/ramses_cc/schemas.py

```diff
@@ -26,313 +25,22 @@
 )
 import voluptuous as vol
 
-from homeassistant.const import ATTR_ENTITY_ID, CONF_SCAN_INTERVAL
+from homeassistant.const import CONF_SCAN_INTERVAL
 from homeassistant.core import callback
 from homeassistant.helpers import config_validation as cv
 
 from .const import (
-    ATTR_ACTIVE,
-    ATTR_CO2_LEVEL,
-    ATTR_COMMAND,
-    ATTR_DELAY_SECS,
-    ATTR_DIFFERENTIAL,
-    ATTR_DURATION,
-    ATTR_INDOOR_HUMIDITY,
-    ATTR_LOCAL_OVERRIDE,
-    ATTR_MAX_TEMP,
-    ATTR_MIN_TEMP,
-    ATTR_MODE,
-    ATTR_MULTIROOM,
-    ATTR_NUM_REPEATS,
-    ATTR_OPENWINDOW,
-    ATTR_OVERRUN,
-    ATTR_PERIOD,
-    ATTR_SCHEDULE,
-    ATTR_SETPOINT,
-    ATTR_TEMPERATURE,
-    ATTR_TIMEOUT,
-    ATTR_UNTIL,
     CONF_ADVANCED_FEATURES,
     CONF_DEV_MODE,
     CONF_MESSAGE_EVENTS,
+    CONF_SEND_PACKET,
     CONF_UNKNOWN_CODES,
-    SVC_DELETE_COMMAND,
-    SVC_FAKE_DEVICE,
-    SVC_FORCE_UPDATE,
-    SVC_GET_DHW_SCHEDULE,
-    SVC_GET_ZONE_SCHEDULE,
-    SVC_LEARN_COMMAND,
-    SVC_PUT_CO2_LEVEL,
-    SVC_PUT_DHW_TEMP,
-    SVC_PUT_INDOOR_HUMIDITY,
-    SVC_PUT_ZONE_TEMP,
-    SVC_RESET_DHW_MODE,
-    SVC_RESET_DHW_PARAMS,
-    SVC_RESET_SYSTEM_MODE,
-    SVC_RESET_ZONE_CONFIG,
-    SVC_RESET_ZONE_MODE,
-    SVC_SEND_COMMAND,
-    SVC_SEND_PACKET,
-    SVC_SET_DHW_BOOST,
-    SVC_SET_DHW_MODE,
-    SVC_SET_DHW_PARAMS,
-    SVC_SET_DHW_SCHEDULE,
-    SVC_SET_SYSTEM_MODE,
-    SVC_SET_ZONE_CONFIG,
-    SVC_SET_ZONE_MODE,
-    SVC_SET_ZONE_SCHEDULE,
-    SystemMode,
-    ZoneMode,
 )
 
 _SchemaT: TypeAlias = dict[str, Any]
 
 _LOGGER = logging.getLogger(__name__)
 
-#
-# Generic services for Integration/domain
-SCH_FAKE_DEVICE = vol.Schema(
-    {
-        vol.Required(SZ_DEVICE_ID): cv.matches_regex(r"^[0-9]{2}:[0-9]{6}$"),
-        vol.Optional("create_device", default=False): vol.Any(None, cv.boolean),
-        vol.Optional("start_binding", default=False): vol.Any(None, cv.boolean),
-    }
-)
```

Comment:
I'm happy to see this go, but I thought this was so elegant:
```python
services = {k: v for k, v in locals().items() if k.startswith("svc")}
```

Regardless, why not leave all the schema definitions in **schemas.py** - seems tidier to me.

Not a biggie - up to you.

Reply:
I'd have perhaps been inclined to leave it, but this did not seem so elegant:
```
if not broker.config[SZ_ADVANCED_FEATURES].get(SVC_SEND_PACKET):
        del domain_service[SVC_SEND_PACKET]
```

I favoured the "don't make me think" approach and split them out as it was easier to follow that way.

Reply:
On the note of `schemas.py` when looking through the core components from what I could see the convention was to have the schemas defined either inline or as a const in the same module so it seemed fairly sensible to follow suit. I suspect it will probably also make more sense later when the main config is fitted into a config flow as it may need to be split up a bit.